scripts/ci/critical_mutation_gate.py

```
@@ -0,0 +1,337 @@
+                            argument
+                            for test_arg in test_args
+                            for argument in ("--cargo-test-arg", test_arg)
+                        ),
+                    ],
+                    repo_root,
+                )
+                if result.returncode not in (0, 2):
+                    raise GateError(
+                        f"cargo-mutants failed for {path} with status {result.returncode}: "
+                        f"{result.stderr.strip() or result.stdout.strip()}"
+                    )
```

Ruff (0.16.0)

[warning] Abstract `raise` to an inner function

(TRY301)